com.unity.ml-agents/Runtime/AccessUtilities.cs

```diff
@@ -0,0 +1,29 @@
+using UnityEngine;
+
+namespace MLAgents
+{
+    public class AccessUtilities
+    {
+        public static bool CanUpdateModelProperties()
+        {
+            return !Application.isPlaying;
```

Comment:
We probably want more complexity than that. If a user wants to create an agent via script for testing, they should be able to modify these properties even if the application is playing no?

Reply:
It's maybe still a little too restrictive, but it's definitely better than before.

I don't know how we can any better than this without taking a major step back, since we don't e.g. if a given RayPerception sensor is attached to an Agent that's already stepping.

Reply:
I agree with @vincentpierre. It doesn't solve the problem if I'm trying to initialize the agent from script. I receive settings at runtime when reading the side channels during Academy Initialization/Awake. But according to the [docs](https://docs.unity3d.com/ScriptReference/Application-isPlaying.html) isPlaying will always be true when called in a built player. So I still cannot set any properties.

What do you think about a new Academy property such as: 
```cs
public static bool hadFirstReset
{
    get { return Instance.m_FirstAcademyReset; }
}
```
You could then just do
```suggestion
            return !Academy.hadFirstReset;
```

From my understanding this will make sure that I cannot set any properties as soon as the Academy was reset for the first time, which happens latest in the first environment step. At this time all Agents should be initialized.

Reply:
Thanks for the feedback. I decided to scrap that approach, since even `Academy.hadFirstReset` would be too restrictive in general (there's no reason why you shoudn't be able to modify properties on an Agent that hasn't been initalized yet).